doc/source/security/data-masking.rst

```diff
@@ -0,0 +1,316 @@
+              | result                               |
+              +--------------------------------------+
+              | 56                                   |
+              +--------------------------------------+
+
+              mysql> SELECT gen_range(-100,-80);
+
+              +--------------------------------------+
+              | gen_range(-100,-80)                  |
+              +--------------------------------------+
+              | -91                                  |
+              +--------------------------------------+
+
+    * - gen_rnd_email()
+      - Generates a random email address. The domain is ``example.com``.
+
+      - .. code-block:: MySQL
+
+             mysql> SELECT gen_rnd_email();
+
+             +---------------------------------------+
+             | gen_rnd_email()                       |
+             +---------------------------------------+
+             | sma.jrts@example.com                  |
+             +---------------------------------------+
+
+    * - gen_rnd_pan([size in integer])
+      - Generates a random primary account number. This function should only
+        be used for test purposes.
+
+      - .. code-block:: MySQL
+
+              mysql> SELECT mask_pan(gen_rnd_pan());
+
+              +-------------------------------------+
+              | mask_pan(gen_rnd_pan())             |
+              +-------------------------------------+
+              | XXXXXXXXXXXX4444                    |
+              +-------------------------------------+
+
+    * - gen_rnd_us_phone()
+      - Generates a random U.S. phone number. The generated number adds the
+        `1` dialing code and is in the `555` area code. The `555` area code
+        is not valid for any U.S. phone number.
+      - .. code-block:: MySQL
+
+            mysql> SELECT gen_rnd_us_phone();
+
+            +-------------------------------+
+            | gen_rnd_us_phone()            |
+            +-------------------------------+
+            | 1-555635-5709                 |
+            +-------------------------------+
+
+    * - gen_rnd_ssn()
+      - Generates a random, non-legitimate US Social Security Number in
+        an ``AAA-BBB-CCCC`` format. This function should only be used for test
+        purposes.
+      - .. code-block:: MySQL
+
+          mysql> SELECT gen_rnd_ssn()
+
+          +-----------------------------+
+          | gen_rnd_ssn()               |
+          +-----------------------------+
+          | 995-33-5656                 |
+          +-----------------------------+
+
+.. rubric:: Using Dictionaries to Generate Random Terms
+
+Data masking returns a value from a range. To use a predefined file as the range to select a string value, load and use a dictionary. A dictionary supports only strings and is loaded from a file with the following characteristics:
+
+* Plain text
+
+* One term per line
+
+* Must contain at least one entry
+
+An example of a dictionary is a list of trees:
```

Comment:
list of trees stored in a file. In our example, it is /usr/local/mysql/dict-files/testdict.

Can you please rephrase?